packages/expansion-hub/src/ExpansionHub.ts

```diff
@@ -96,44 +97,123 @@ export interface ExpansionHub extends RevHub {
     getI2CReadStatus(i2cChannel: number): Promise<I2CReadStatus>;
 
     // Motor
+    /**
+     * Configure a specific motor
+     * @param motorChannel
+     * @param motorMode
+     * @param floatAtZero whether to coast when power is set to 0. If this is set to true, the motor will brake at 0 instead.
+     */
     setMotorChannelMode(
         motorChannel: number,
-        motorMode: number,
+        motorMode: MotorMode,
         floatAtZero: boolean,
     ): Promise<void>;
+
+    /**
+     *
+     * Get the configuration for a specific motor
+     * @param motorChannel
+     */
     getMotorChannelMode(
         motorChannel: number,
-    ): Promise<{ motorMode: number; floatAtZero: boolean }>;
+    ): Promise<{ motorMode: MotorMode; floatAtZero: boolean }>;
+
+    /**
+     * Enable a motor. The motor must be configured before enabling.
+     * @param motorChannel
+     * @param enable
+     */
+
     setMotorChannelEnable(motorChannel: number, enable: boolean): Promise<void>;
+
+    /**
+     * Check whether a motor channel is currently enabled
+     * @param motorChannel
+     */
     getMotorChannelEnable(motorChannel: number): Promise<boolean>;
     setMotorChannelCurrentAlertLevel(
         motorChannel: number,
         currentLimit_mA: number,
     ): Promise<void>;
+
+    /**
+     * Get the present current alert (mA)
+     * @param motorChannel
+     */
     getMotorChannelCurrentAlertLevel(motorChannel: number): Promise<number>;
     resetMotorEncoder(motorChannel: number): Promise<void>;
+
+    /**
+     * Set the power for a motor
+     * @param motorChannel the motor
+     * @param powerLevel power in range [-1,1]
+     */
     setMotorConstantPower(motorChannel: number, powerLevel: number): Promise<void>;
+
+    /**
+     * Get the current constant power setting for a motor
+     * @param motorChannel
+     */
     getMotorConstantPower(motorChannel: number): Promise<number>;
+
+    /**
+     * Set the target velocity for a motor
+     * @param motorChannel the motor
+     * @param velocity_cps velocity in encoder counts per second
+     */
     setMotorTargetVelocity(motorChannel: number, velocity_cps: number): Promise<void>;
+
+    /**
+     * Get the current target velocity for a motor
+     * @param motorChannel
+     */
     getMotorTargetVelocity(motorChannel: number): Promise<number>;
     setMotorTargetPosition(
         motorChannel: number,
         targetPosition_counts: number,
         targetTolerance_counts: number,
     ): Promise<void>;
+
+    /**
+     * Get the current target position for a motor
+     * @param motorChannel
+     */
     getMotorTargetPosition(
         motorChannel: number,
     ): Promise<{ targetPosition: number; targetTolerance: number }>;
+
+    /**
+     * Check if a motor has reached its target
```

Comment:
Many of these methods need documentation that explains how they interact with the current motor mode. For example, this method is only applicable to RUN_TO_POSITION mode.

Reply:
Added note where applicable